core/sigagg/sigagg.go

```diff
@@ -89,3 +95,25 @@ func (a *Aggregator) Aggregate(ctx context.Context, duty core.Duty, pubkey core.
 
 	return nil
 }
+
+// verifyAggSig verifies if the aggregated signature is valid using the provided pubkey.
+func verifyAggSig(sigData core.SignedData, pubkey core.PubKey) error {
+	tblsPubkey, err := tblsconv.PubkeyFromCore(pubkey)
+	if err != nil {
+		return errors.Wrap(err, "pubkey from core")
+	}
+
+	msgRoot, err := sigData.MessageRoot()
+	if err != nil {
+		return errors.Wrap(err, "message root")
+	}
+
+	aggSig := tbls.Signature(sigData.Signature())
+
+	err = tbls.Verify(tblsPubkey, msgRoot[:], aggSig)
```

Comment:
this is a eth2 signature, so it should include eth2 domains. I do not want to add a eth2client to sigagg, so I would actually mark this as WONTFIX.

Reasoning:
  - We already validate the partial signatures.
  - So if aggregation succeeds then resulting aggregate signature is also valid. 
  - Additional validation isn't required.

Reply:
If invalid aggregate signatures are however still possible given valid partial signatures and succesful aggregation, then we should follow the parsigex approach of decoupling validation from the component via a `verifyFunc`.

Reply:
See parsigex how to verify eth2 signatures

Reply:
I would say we err on the side of caution here and still validate aggregated signatures. I'll follow the `verifyFunc` approach of verifying signatures